pyproject.toml

```diff
@@ -95,6 +98,82 @@ push = true
     '^{version}$'
 ]
 
+[tool.pylint.main]
+py-version = "3.7"
+init-hook = 'import sys; sys.path.append("scripts/thirdparties")'
+
+load-plugins = [
+    "pylint.extensions.bad_builtin"
+]
+
+# If pylint fails, check that the code is rated higher than `fail-under`,
+# and search for messages from `fail-on`.
+fail-under = 9.9
+fail-on = [
+    # Messages with numerical IDs `E\d+` or `F\d+`.
+    "E", "F",
+    # Use logging framework as much as possible instead of print().
+    "bad-builtin",
+    # https://github.com/pyinstaller/pyinstaller/issues/1687
+    "consider-using-sys-exit",
+    # Considered more error prone because we have __main__ startup dependent on the initial procedure.
+    "cyclic-import",
+    # Since we use ruamel.yaml, using of `type() == ` checks is considered more error prone.
+    "unidiomatic-typecheck",
+    # Consider using kubemarine.core.utils.open_(external|internal).
+    "unspecified-encoding",
+    # List of other warnings that are considered more error prone
+    "arguments-differ",
+    "arguments-renamed",
+    "cell-var-from-loop",
+    "implicit-str-concat",
+    "undefined-loop-variable",
+    # List of other non-critical warnings that rarely appear
+    "useless-parent-delegation",
+]
```

Comment:
Currently, pylint output is clear.
`fail-under = 9.9` allows some messages, but the CI job will eventually fail after the code is spoiled.
Alternatives are:
* Fail if any message is printed. This will make development more complex.
* Fail for any warning. Less complex, but some warnings are still quite unimportant, e.g. `useless-else-on-loop`.
* Do not use fail-under. The cons is as long as the output rise, it will become harder to find messages that really caused the failure, e.g. `consider-using-sys-exit`.

Reply:
Decided to fail if any message is printed.